dissect/tests/test_fat32.py

```diff
@@ -0,0 +1,434 @@
+        fs.bpb.BPB_RootClus = 2
+        fs.bpb.BPB_FSInfo = 1
+        fs.bpb.BPB_BkBootSec = 6
+        fs.bpb.BPB_DrvNum = 128  # from Linux example
+        fs.bpb.BPB_Reserved1 = 0
+        fs.bpb.BPB_BootSig = 41
+        fs.bpb.BPB_VolID = 4107516940  # from Linux example
+        fs.bpb.BPB_VolLab = b"NO NAME    "
+        fs.bpb.BPB_FilSysType = b"FAT32   "
+        fs.bpb.BPB_BootCode = NOBOOT_CODE
+        fs.bpb.EndOfSectorMarker = 0xAA55
+
+
+    # the the FS is first parsed, it may not create structures for the
+    # backup BPB, FS info, or FATs, since they're not referenced in a NULL primary BPB.
+    # so, we reload the FS using the initial BPB set just above.
+    with open(path, "r+b") as f:
+        fs.vsLoad(f, partition_offset, writeback=True)
+
+        fs.bpb_backup.BPB_jmpBoot = fs.bpb.BPB_jmpBoot
+        fs.bpb_backup.BPB_OEMName = fs.bpb.BPB_OEMName
+        fs.bpb_backup.BPB_BytsPerSec = fs.bpb.BPB_BytsPerSec
+        fs.bpb_backup.BPB_SecPerClus = fs.bpb.BPB_SecPerClus
+        fs.bpb_backup.BPB_RsvdSecCnt = fs.bpb.BPB_RsvdSecCnt
+        fs.bpb_backup.BPB_NumFATs = fs.bpb.BPB_NumFATs
+        fs.bpb_backup.BPB_RootEntCnt = fs.bpb.BPB_RootEntCnt
+        fs.bpb_backup.BPB_TotSec16 = fs.bpb.BPB_TotSec16
+        fs.bpb_backup.BPB_TotSec32 = fs.bpb.BPB_TotSec32
+        fs.bpb_backup.BPB_Media = fs.bpb.BPB_Media
+        fs.bpb_backup.BPB_FATSz16 = fs.bpb.BPB_FATSz16
+        fs.bpb_backup.BPB_SecPerTrk = fs.bpb.BPB_SecPerTrk
+        fs.bpb_backup.BPB_NumHeads = fs.bpb.BPB_NumHeads
+        fs.bpb_backup.BPB_HiddSec = fs.bpb.BPB_HiddSec
+        fs.bpb_backup.BPB_FATSz32 = fs.bpb.BPB_FATSz32
+        fs.bpb_backup.BPB_ExtFlags = fs.bpb.BPB_ExtFlags
+        fs.bpb_backup.BPB_FSVer = fs.bpb.BPB_FSVer
+        fs.bpb_backup.BPB_RootClus = fs.bpb.BPB_RootClus
+        fs.bpb_backup.BPB_FSInfo = fs.bpb.BPB_FSInfo
+        fs.bpb_backup.BPB_BkBootSec = fs.bpb.BPB_BkBootSec
+        fs.bpb_backup.BPB_DrvNum = fs.bpb.BPB_DrvNum
+        fs.bpb_backup.BPB_Reserved1 = fs.bpb.BPB_Reserved1
+        fs.bpb_backup.BPB_BootSig = fs.bpb.BPB_BootSig
+        fs.bpb_backup.BPB_VolID = fs.bpb.BPB_VolID
+        fs.bpb_backup.BPB_VolLab = fs.bpb.BPB_VolLab
+        fs.bpb_backup.BPB_FilSysType = fs.bpb.BPB_FilSysType
+        fs.bpb_backup.BPB_BootCode = fs.bpb.BPB_BootCode
+        fs.bpb_backup.EndOfSectorMarker = fs.bpb.EndOfSectorMarker
+
+        for f in fs.fats:
+            f[0] = 0x0FFFFFF8  # from Linux sample
+            f[1] = 0x0FFFFFFF
+
+            # set root dir cluster chain
+            f[2] = fat32.CLUSTER_TYPES.LAST
+
+        fs.fs_info.FSI_LeadSig = 0x41615252
+        fs.fs_info.FSI_StrucSig = 0x61417272
+        fs.fs_info.FSI_TailSig = 0xAA550000
+
+
+@contextlib.contextmanager
+def test_image():
+    try:
+        os.remove(TEST_IMAGE_PATH)
+    except:
+        pass
+
+    with open(TEST_IMAGE_PATH, "wb") as f:
+        f.write(b"\x00" * TEST_IMAGE_SIZE)
+
+    try:
+        yield
+
```

Comment:
exactly. now, it seems i've commented out the `os.remove(...)` so I'll have to fix this. this is essentially the same as having a `try... finally` block, but much cleaner (since you only describe the setup/cleanup once).